### ISSUE TITLE
Make validation errors easier to find

### ISSUE THREAD

**Issue by sweetgiorni** [json-editor/json-editor#845]
I use a schema with a "tabs" formatted array. I was having problems where users would have tab B open, so they  wouldn't be able to see the validation errors in tab A. I temporarily solved it with an ugly  table at the bottom of the page showing the error list, but it would be nice to have an option that makes them easier to find. Here are some potential options:

1. When validation occurs and an error is in another tab, automatically open that tab
2. When validation occurs and an error is in another tab, highlight the tab red
3. When validation occurs and the invalid field is not in view, use scrollIntoView() to make the it visible on the screen

I can work on this enhancement if it's wanted.

**Comment by schmunk42** [comment]
I like the summary very much, too. 
CC: @marc7000 @germanbisurgi We already have error message boxes above certain types, in which cases do they occur?

Option 2 would be my favourite, "simply" adding a css class to the tab.

**Comment by sweetgiorni** [comment]
Yeah the summary table might work with some changes. I suppose the biggest issue (besides the ugliness of my implementation) is the list of field paths. It's a bit weird looking to the end-user:

![image](https://user-images.githubusercontent.com/12527888/96798290-6afe6f80-13b5-11eb-8a07-5e4ce0f0da4f.png)

Maybe I could replace those field paths with links that take you to the bad field?

**Comment by schmunk42** [comment]
It would be great if we could have a template for this, where you could ie. also use `Field Title` or `Field Description`.

Ie. I would also rather use a list than a table by default.


(pseudo code)
```
editor.config.options.errorSummary = "
 <div class="error-summary>
   <ul>
     (foreach...)
     <li>{{ field.title }} {{ field.error }}</li>
     (endforeach) 
   </ul> 
 </div>
"
```

see also https://github.com/json-editor/json-editor#templates

But the rendering should be an editor config option, not something in the schema.

CC: @germanbisurgi @marc7000

**Comment by PythonLinks** [comment]
I need this.
I like making the tabs red. 
Your validation errors table would be more useful if it first said the tab name, then the field name. 
But once you did that, probably not too hard to change the tab color. 
Of course the tricky part is that on every event, you have to figure out if the red color goes away.

**Comment by Delwing** [comment]
Table is cool, but it would be really cool to have validation error event.
That way custom summary could be made, scrolling from error to error etc.

**Comment by eduo** [comment]
I know this is a bit in the air for some time. But I was looking at ticket #1245 and if sections could be expanded/collapses programmatically it would be easy to parse the path.

Today if there's no section collapsed it's easy to do something like this:

editor.getEditor('root.objetos.0.proveedor.dom.Numero').input.scrollIntoViewIfNeeded()

But if sections are collapsed this won't work, as items are invisible.

editor.getEditor('root.objetos.0.proveedor.dom.Numero').display() could make sure the hierarchy is visible if it isn't, without expanding the whole form.

alternatively

editor.validate()[0].scrollIntoView() could both expand and scroll.

I would prefer the first one as I could attach a highlight animation to the field, useful for busy forms.
